### PR TITLE
[RCM-499] Send trace agent env

### PR DESCRIPTION
**Pull request by coignetp**
### What does this PR do?

Send trace agent environment in the RC request, so the backend can filter on that in addition to tracer environment that can be different

Move the logic for `DefaultEnv` to a common location

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by BaptisteFoy — APPROVED** [review]

**Review by djmitche — APPROVED** [review]

**Review by nizox — APPROVED** [review]

**Review by ajgajg1134 — APPROVED** [review]
Looks good from Agent-APM!

File: cmd/trace-agent/config/config.go [hunk 1]
```diff
@@ -180,21 +180,12 @@ func applyDatadogConfig(c *config.AgentConfig) error {
...
-				log.Debugf("Setting DefaultEnv to %q (from `env:` entry under the 'tags' config option: %q)", c.DefaultEnv, tag)
-				break
-			}
-		}
+
+	newDefaultEnv := coreconfig.GetTraceAgentDefaultEnv()
+	if newDefaultEnv != "" {
+		c.DefaultEnv = newDefaultEnv
 	}
+
```
Comment (katiehockman) [inline cmd/trace-agent/config/config.go:188]:
```suggestion
	if env := coreconfig.GetTraceAgentDefaultEnv(); env != "" {
		c.DefaultEnv = newDefaultEnv
	}

```

File: pkg/config/config.go [hunk 2]
```diff
@@ -1888,3 +1888,25 @@ func GetVectorURL(datatype DataType) (string, error) {
 	}
 	return "", nil
 }
+
+// GetTraceAgentDefaultEnv returns the default env for the trace agent
+func GetTraceAgentDefaultEnv() string {
```
Comment (katiehockman) [inline pkg/config/config.go:1893]: nit: now that it's in a separate function, I'd just return the value at every point that you call `defaultEnv = `. It'll make the logic a bit simpler.

e.g. line 1896 just do `return Datadog.GetString("apm_config.env")`

**Review by katiehockman — APPROVED** [review]
Just a few nits

File: pkg/config/config.go (hunk 2 above)
Comment (coignetp) [reply pkg/config/config.go:1893]: We still need to log where the environment comes from in the config, I'm not sure it will make the logic simpler

Comment (katiehockman) [reply pkg/config/config.go:1893]: Oh you can just log before you return. That's totally fine.

**Pull request merged**